src/ooxml/java/org/apache/poi/openxml4j/opc/PackagePart.java

```diff
@@ -457,7 +460,16 @@ public boolean hasRelationships() {
 	 * @see org.apache.poi.openxml4j.opc.RelationshipSource#isRelationshipExists(org.apache.poi.openxml4j.opc.PackageRelationship)
 	 */
 	public boolean isRelationshipExists(PackageRelationship rel) {
-		return _relationships.getRelationshipByID(rel.getId()) != null;
```

Comment:
for some reason the id might be null - the code should now throw up on this. The patch catches the original problem further up the code hierarchy but it took me to find out because of the null pointer exceptions further down the hierarchy.